src/Compilers/Core/Portable/Syntax/GreenNode.cs

```diff
@@ -541,17 +560,15 @@ private static IEnumerable<SyntaxAnnotation> GetAnnotationsSlow(SyntaxAnnotation
 
         public SyntaxAnnotation[] GetAnnotations()
         {
-            if (this.ContainsAnnotations)
-            {
-                SyntaxAnnotation[]? annotations;
-                if (s_annotationsTable.TryGetValue(this, out annotations))
-                {
-                    System.Diagnostics.Debug.Assert(annotations.Length != 0, "we should return nonempty annotations or NoAnnotations");
-                    return annotations;
-                }
-            }
```

Comment:
this is the meat of the change.  In the past, we'd end up lookign at nodes that didn't actually have annotations on htem, because they had a child somewhere deep under them with annotations.  This was very expensive as CWT lookups are not cheap.  For tree rewrites, when rewriting nodes , we look at the old node to copy annotations over.  And this meant almost always doing this work just if a deep child token had an annotation.  now we only need to do the work on nodes that actually have annotations on them themselves.